scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryPartitionUtil.scala

```diff
@@ -0,0 +1,97 @@
+package com.spotify.scio.bigquery
+
+import java.util.regex.Pattern
+
+import com.google.api.services.bigquery.model.TableReference
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO
+
+private[bigquery] object BigQueryPartitionUtil {
+
+  // Ported from com.google.cloud.dataflow.sdk.io.BigQueryIO
+
+  private val PROJECT_ID_REGEXP = "[a-z][-a-z0-9:.]{4,61}[a-z0-9]"
+  private val DATASET_REGEXP = "[-\\w.]{1,1024}"
+  private val TABLE_REGEXP = "[-\\w$@]{1,1024}($LATEST)?"
+  private val DATASET_TABLE_REGEXP_LEGACY =
+    s"((?<PROJECT>$PROJECT_ID_REGEXP):)?(?<DATASET>$DATASET_REGEXP)\\.(?<TABLE>$TABLE_REGEXP)"
+  private val DATASET_TABLE_REGEXP_STANDARD =
+    s"((?<PROJECT>$PROJECT_ID_REGEXP).)?(?<DATASET>$DATASET_REGEXP)\\.(?<TABLE>$TABLE_REGEXP)"
+  private val QUERY_TABLE_SPEC_LEGACY =
+    Pattern.compile(s"(?<=\\[)$DATASET_TABLE_REGEXP_LEGACY(?=\\])")
+  private val QUERY_TABLE_SPEC_STANDARD =
+    Pattern.compile(s"(?<=\\`)$DATASET_TABLE_REGEXP_STANDARD(?=\\`)")
+
+  private def extractTables(sqlQuery: String): Map[String, TableReference] = {
+    val b = Map.newBuilder[String, TableReference]
+    val m1 = QUERY_TABLE_SPEC_LEGACY.matcher(sqlQuery)
+    while (m1.find()) {
+      val t = m1.group(0)
+      b += (s"[$t]" -> BigQueryIO.parseTableSpec(t))
+    }
+    val m2 = QUERY_TABLE_SPEC_STANDARD.matcher(sqlQuery)
+    while (m2.find()) {
+      val t = m2.group(0)
+      b += (s"`$t`" -> BigQueryIO.parseTableSpec(t.replaceFirst("\\.", ":")))
+    }
+    b.result()
+  }
+
+  private def getPartitions(bq: BigQueryClient, tableRef: TableReference): Set[String] = {
```

Comment:
isn't partition in bigquery is another dimension of the table. maybe getTableIds will be a nicer name?